v1/tarball/image.go

```diff
@@ -177,6 +181,94 @@ func (i *image) RawConfigFile() ([]byte, error) {
 	return i.config, nil
 }
 
+// Flatten takes an image and flattens its layers into a single tarball at the specified path
+// Adapted from https://github.com/google/containerregistry/blob/master/client/v2_2/docker_image_.py#L731
+func Flatten(img v1.Image, tarFilePath string) error {
+	tarFile, err := os.Create(tarFilePath)
+	if err != nil {
+		return fmt.Errorf("Error creating tar file: %v", err)
+	}
+	defer tarFile.Close()
+	tarWriter := tar.NewWriter(tarFile)
+	defer tarWriter.Close()
+
+	fileMap := make(map[string]bool, 0)
```

Comment:
nit: I find the map literal form a bit more straightforward: fileMap := map[string]bool{}